From 27515371fa2be1df0e1074e7dac8a627da3ca6af Mon Sep 17 00:00:00 2001
From: Fabrice Fontaine <fontaine.fabrice@gmail.com>
Date: Sat, 11 Apr 2020 19:27:06 +0200
Subject: [PATCH] package/open-lldp: bump version to
 b71bfb87fefb31c4b1a6a7ae351791c90966c3a8

- Switch site to github
- Remove first patch (already in version)
- Update second patch
- Update indentation of hash file (two spaces)

Signed-off-by: Fabrice Fontaine <fontaine.fabrice@gmail.com>
Signed-off-by: Thomas Petazzoni <thomas.petazzoni@bootlin.com>
---
 ...-functions-work-with-both-gnu11-and-gnu89.patch | 52 +++++++++++++++
 ...1-vdptool-fixed-compile-error-for-getline.patch | 27 --------
 .../0002-Makefile.am-disable-Werror.patch          | 38 +++++++++++
 ...-functions-work-with-both-gnu11-and-gnu89.patch | 73 ----------------------
 .../0003-Makefile.am-disable-Werror.patch          | 38 -----------
 .../open-lldp/0003-fixed-unknown-type-pid_t.patch  | 42 +++++++++++++
 .../open-lldp/0004-fixed-unknown-type-pid_t.patch  | 41 ------------
 package/open-lldp/Config.in                        |  2 +-
 package/open-lldp/open-lldp.hash                   |  4 +-
 package/open-lldp/open-lldp.mk                     |  4 +-
 10 files changed, 137 insertions(+), 184 deletions(-)
 create mode 100644 package/open-lldp/0001-get-inline-functions-work-with-both-gnu11-and-gnu89.patch
 delete mode 100644 package/open-lldp/0001-vdptool-fixed-compile-error-for-getline.patch
 create mode 100644 package/open-lldp/0002-Makefile.am-disable-Werror.patch
 delete mode 100644 package/open-lldp/0002-get-inline-functions-work-with-both-gnu11-and-gnu89.patch
 delete mode 100644 package/open-lldp/0003-Makefile.am-disable-Werror.patch
 create mode 100644 package/open-lldp/0003-fixed-unknown-type-pid_t.patch
 delete mode 100644 package/open-lldp/0004-fixed-unknown-type-pid_t.patch

diff --git a/package/open-lldp/0001-get-inline-functions-work-with-both-gnu11-and-gnu89.patch b/package/open-lldp/0001-get-inline-functions-work-with-both-gnu11-and-gnu89.patch
new file mode 100644
index 0000000..a0980e8
--- /dev/null
+++ b/package/open-lldp/0001-get-inline-functions-work-with-both-gnu11-and-gnu89.patch
@@ -0,0 +1,52 @@
+From e186ca9ebda181f62504a1922559ef3ddae5e3b6 Mon Sep 17 00:00:00 2001
+From: Laurent Charpentier <laurent_pubs@yahoo.com>
+Date: Mon, 15 Jan 2018 11:24:45 +0100
+Subject: [PATCH] get inline functions work with both gnu11 and gnu89
+
+After gcc upgraded to gcc5, and if the codes are compiled without optimization(-O0), and the below error will happen:
+
+./include/lldp_8021qaz.h:237:12: error: inline function 'ieee8021qaz_clif_cmd' declared but never defined [-Werror]
+ inline int ieee8021qaz_clif_cmd(void *data, struct sockaddr_un *from,
+            ^
+./include/lldp_8021qaz.h:222:13: error: inline function 'set_prio_map' declared but never defined [-Werror]
+ inline void set_prio_map(u32 *prio_map, u8 prio, int tc);
+             ^
+./include/lldp_8021qaz.h:221:12: error: inline function 'get_prio_map' declared but never defined [-Werror]
+ inline int get_prio_map(u32 prio_map, int tc);
+
+gcc5 defaults to -std=gnu11 instead of -std=gnu89, and it requires that exactly one C source file has the callable copy of the inline function.
+
+Signed-off-by: Laurent Charpentier <laurent_pubs@yahoo.com>
+[Fabrice: patch updated so it applies after
+https://github.com/intel/openlldp/commit/7ad7a2f2adc70d8e9213f8bf511bf3374d64b034]
+Signed-off-by: Fabrice Fontaine <fontaine.fabrice@gmail.com>
+---
+ include/lldp_8021qaz.h | 6 ------
+ lldp_8021qaz.c         | 4 ++--
+ 2 files changed, 2 insertions(+), 8 deletions(-)
+
+diff --git a/lldp_8021qaz.c b/lldp_8021qaz.c
+index 094676d..f154317 100644
+--- a/lldp_8021qaz.c
++++ b/lldp_8021qaz.c
+@@ -396,7 +396,7 @@ static int read_cfg_file(char *ifname, struct lldp_agent *agent,
+ 	return 0;
+ }
+ 
+-inline int get_prio_map(u32 prio_map, int prio)
++static inline int get_prio_map(u32 prio_map, int prio)
+ {
+ 	if (prio > 7)
+ 		return 0;
+@@ -404,7 +404,7 @@ inline int get_prio_map(u32 prio_map, int prio)
+ 	return (prio_map >> (4 * (7-prio))) & 0xF;
+ }
+ 
+-inline void set_prio_map(u32 *prio_map, u8 prio, int tc)
++static inline void set_prio_map(u32 *prio_map, u8 prio, int tc)
+ {
+ 	u32 mask = ~(0xffffffff & (0xF << (4 * (7-prio))));
+ 	*prio_map &= mask;
+-- 
+2.14.3
+
diff --git a/package/open-lldp/0001-vdptool-fixed-compile-error-for-getline.patch b/package/open-lldp/0001-vdptool-fixed-compile-error-for-getline.patch
deleted file mode 100644
index c202677..0000000
--- a/package/open-lldp/0001-vdptool-fixed-compile-error-for-getline.patch
+++ /dev/null
@@ -1,27 +0,0 @@
-From 3c7887bb8a61a3b49a2989329c756a038023c544 Mon Sep 17 00:00:00 2001
-From: Laurent Charpentier <laurent_pubs@yahoo.com>
-Date: Fri, 12 Jan 2018 12:25:41 +0100
-Subject: [PATCH] vdptool: fixed compile error for getline()
-
-vdptool.c was generating an error: implicit declaration for the getline() function.
-
-Signed-off-by: Laurent Charpentier <laurent_pubs@yahoo.com>
----
- vdptool.c | 1 +
- 1 file changed, 1 insertion(+)
-
-diff --git a/vdptool.c b/vdptool.c
-index b805372..4930f6a 100644
---- a/vdptool.c
-+++ b/vdptool.c
-@@ -36,6 +36,7 @@
-  * set and query VSI profile settings.
-  */
- 
-+#define _GNU_SOURCE
- #include <stdlib.h>
- #include <stdio.h>
- #include <unistd.h>
--- 
-2.14.3
-
diff --git a/package/open-lldp/0002-Makefile.am-disable-Werror.patch b/package/open-lldp/0002-Makefile.am-disable-Werror.patch
new file mode 100644
index 0000000..d6393bd
--- /dev/null
+++ b/package/open-lldp/0002-Makefile.am-disable-Werror.patch
@@ -0,0 +1,38 @@
+From c350f263136801fd54d42322a07a3b5ad2181dd9 Mon Sep 17 00:00:00 2001
+From: Thomas Petazzoni <thomas.petazzoni@free-electrons.com>
+Date: Thu, 18 Jan 2018 09:29:32 +0100
+Subject: [PATCH] Makefile.am: disable -Werror
+
+Due to additional warnings in recent gcc versions and the use of
+-Werror the build fails for reasons that are not that important. For
+example:
+
+clif.c:54:2: error: implicit declaration of function 'memset' [-Werror=implicit-function-declaration]
+  memset(clif, 0, sizeof(*clif));
+
+or:
+
+log.c:46:30: error: format '%ld' expects argument of type 'long int', but argument 5 has type '__suseconds_t {aka int}' [-Werror=format=]
+   printf("%02d:%02d:%02d.%06ld ",
+
+Signed-off-by: Thomas Petazzoni <thomas.petazzoni@free-electrons.com>
+---
+ Makefile.am | 2 +-
+ 1 file changed, 1 insertion(+), 1 deletion(-)
+
+diff --git a/Makefile.am b/Makefile.am
+index 551d4c7..62a52a5 100644
+--- a/Makefile.am
++++ b/Makefile.am
+@@ -16,7 +16,7 @@ ACLOCAL_AMFLAGS = -I m4
+ parse_cli.o: CFLAGS+=-U_FORTIFY_SOURCE -Wno-error
+ 
+ ## system requires a shared libconfig
+-AM_CFLAGS = -Wall -Werror -Wextra -Wformat=2 $(LIBCONFIG_CFLAGS) $(LIBNL_CFLAGS)
++AM_CFLAGS = -Wall -Wextra -Wformat=2 $(LIBCONFIG_CFLAGS) $(LIBNL_CFLAGS)
+ AM_LDFLAGS = $(LIBCONFIG_LIBS) $(LIBNL_LIBS) -lrt
+ 
+ ## header files to be installed, for programs using the client interface to lldpad 
+-- 
+2.14.3
+
diff --git a/package/open-lldp/0002-get-inline-functions-work-with-both-gnu11-and-gnu89.patch b/package/open-lldp/0002-get-inline-functions-work-with-both-gnu11-and-gnu89.patch
deleted file mode 100644
index 57a23b2..0000000
--- a/package/open-lldp/0002-get-inline-functions-work-with-both-gnu11-and-gnu89.patch
+++ /dev/null
@@ -1,73 +0,0 @@
-From e186ca9ebda181f62504a1922559ef3ddae5e3b6 Mon Sep 17 00:00:00 2001
-From: Laurent Charpentier <laurent_pubs@yahoo.com>
-Date: Mon, 15 Jan 2018 11:24:45 +0100
-Subject: [PATCH] get inline functions work with both gnu11 and gnu89
-
-After gcc upgraded to gcc5, and if the codes are compiled without optimization(-O0), and the below error will happen:
-
-./include/lldp_8021qaz.h:237:12: error: inline function 'ieee8021qaz_clif_cmd' declared but never defined [-Werror]
- inline int ieee8021qaz_clif_cmd(void *data, struct sockaddr_un *from,
-            ^
-./include/lldp_8021qaz.h:222:13: error: inline function 'set_prio_map' declared but never defined [-Werror]
- inline void set_prio_map(u32 *prio_map, u8 prio, int tc);
-             ^
-./include/lldp_8021qaz.h:221:12: error: inline function 'get_prio_map' declared but never defined [-Werror]
- inline int get_prio_map(u32 prio_map, int tc);
-
-gcc5 defaults to -std=gnu11 instead of -std=gnu89, and it requires that exactly one C source file has the callable copy of the inline function.
-
-Signed-off-by: Laurent Charpentier <laurent_pubs@yahoo.com>
----
- include/lldp_8021qaz.h | 6 ------
- lldp_8021qaz.c         | 4 ++--
- 2 files changed, 2 insertions(+), 8 deletions(-)
-
-diff --git a/include/lldp_8021qaz.h b/include/lldp_8021qaz.h
-index 55353b8..09dee20 100644
---- a/include/lldp_8021qaz.h
-+++ b/include/lldp_8021qaz.h
-@@ -218,9 +218,6 @@ int ieee8021qaz_mod_app(struct app_tlv_head *head, int peer,
- 			u8 prio, u8 sel, u16 proto, u32 ops);
- int ieee8021qaz_app_sethw(char *ifname, struct app_tlv_head *head);
- 
--inline int get_prio_map(u32 prio_map, int tc);
--inline void set_prio_map(u32 *prio_map, u8 prio, int tc);
--
- struct ieee8021qaz_tlvs *ieee8021qaz_data(const char *);
- 
- int ieee8021qaz_tlvs_rxed(const char *ifname);
-@@ -234,9 +231,6 @@ int ieee8021qaz_rchange(struct port *port, struct lldp_agent *,
- void ieee8021qaz_ifup(char *ifname, struct lldp_agent *);
- void ieee8021qaz_ifdown(char *ifname, struct lldp_agent *);
- u8 ieee8021qaz_mibDeleteObject(struct port *port, struct lldp_agent *);
--inline int ieee8021qaz_clif_cmd(void *data, struct sockaddr_un *from,
--				socklen_t fromlen, char *ibuf, int ilen,
--				char *rbuf);
- int ieee8021qaz_check_operstate(void);
- int get_dcbx_hw(const char *ifname, __u8 *dcbx);
- 
-diff --git a/lldp_8021qaz.c b/lldp_8021qaz.c
-index 094676d..f154317 100644
---- a/lldp_8021qaz.c
-+++ b/lldp_8021qaz.c
-@@ -396,7 +396,7 @@ static int read_cfg_file(char *ifname, struct lldp_agent *agent,
- 	return 0;
- }
- 
--inline int get_prio_map(u32 prio_map, int prio)
-+static inline int get_prio_map(u32 prio_map, int prio)
- {
- 	if (prio > 7)
- 		return 0;
-@@ -404,7 +404,7 @@ inline int get_prio_map(u32 prio_map, int prio)
- 	return (prio_map >> (4 * (7-prio))) & 0xF;
- }
- 
--inline void set_prio_map(u32 *prio_map, u8 prio, int tc)
-+static inline void set_prio_map(u32 *prio_map, u8 prio, int tc)
- {
- 	u32 mask = ~(0xffffffff & (0xF << (4 * (7-prio))));
- 	*prio_map &= mask;
--- 
-2.14.3
-
diff --git a/package/open-lldp/0003-Makefile.am-disable-Werror.patch b/package/open-lldp/0003-Makefile.am-disable-Werror.patch
deleted file mode 100644
index d6393bd..0000000
--- a/package/open-lldp/0003-Makefile.am-disable-Werror.patch
+++ /dev/null
@@ -1,38 +0,0 @@
-From c350f263136801fd54d42322a07a3b5ad2181dd9 Mon Sep 17 00:00:00 2001
-From: Thomas Petazzoni <thomas.petazzoni@free-electrons.com>
-Date: Thu, 18 Jan 2018 09:29:32 +0100
-Subject: [PATCH] Makefile.am: disable -Werror
-
-Due to additional warnings in recent gcc versions and the use of
--Werror the build fails for reasons that are not that important. For
-example:
-
-clif.c:54:2: error: implicit declaration of function 'memset' [-Werror=implicit-function-declaration]
-  memset(clif, 0, sizeof(*clif));
-
-or:
-
-log.c:46:30: error: format '%ld' expects argument of type 'long int', but argument 5 has type '__suseconds_t {aka int}' [-Werror=format=]
-   printf("%02d:%02d:%02d.%06ld ",
-
-Signed-off-by: Thomas Petazzoni <thomas.petazzoni@free-electrons.com>
----
- Makefile.am | 2 +-
- 1 file changed, 1 insertion(+), 1 deletion(-)
-
-diff --git a/Makefile.am b/Makefile.am
-index 551d4c7..62a52a5 100644
---- a/Makefile.am
-+++ b/Makefile.am
-@@ -16,7 +16,7 @@ ACLOCAL_AMFLAGS = -I m4
- parse_cli.o: CFLAGS+=-U_FORTIFY_SOURCE -Wno-error
- 
- ## system requires a shared libconfig
--AM_CFLAGS = -Wall -Werror -Wextra -Wformat=2 $(LIBCONFIG_CFLAGS) $(LIBNL_CFLAGS)
-+AM_CFLAGS = -Wall -Wextra -Wformat=2 $(LIBCONFIG_CFLAGS) $(LIBNL_CFLAGS)
- AM_LDFLAGS = $(LIBCONFIG_LIBS) $(LIBNL_LIBS) -lrt
- 
- ## header files to be installed, for programs using the client interface to lldpad 
--- 
-2.14.3
-
diff --git a/package/open-lldp/0003-fixed-unknown-type-pid_t.patch b/package/open-lldp/0003-fixed-unknown-type-pid_t.patch
new file mode 100644
index 0000000..9391a04
--- /dev/null
+++ b/package/open-lldp/0003-fixed-unknown-type-pid_t.patch
@@ -0,0 +1,42 @@
+From 41e9e6b5aa13d65480cc960a7f15f97d74d64090 Mon Sep 17 00:00:00 2001
+From: Laurent Charpentier <laurent_pubs@yahoo.com>
+Date: Mon, 22 Jan 2018 10:49:45 +0100
+Subject: [PATCH] fixed unknown type pid_t
+
+Fixed 'unknown type pid_t' gcc compile error in dcbtool_cmds.c and
+lldptool_cmds.c
+
+Signed-off-by: Laurent Charpentier <laurent_pubs@yahoo.com>
+[Upstream status: https://github.com/intel/openlldp/pull/46]
+---
+ dcbtool_cmds.c  | 1 +
+ lldptool_cmds.c | 1 +
+ 2 files changed, 2 insertions(+)
+
+diff --git a/dcbtool_cmds.c b/dcbtool_cmds.c
+index a5cd0fe..ee5c144 100644
+--- a/dcbtool_cmds.c
++++ b/dcbtool_cmds.c
+@@ -27,6 +27,7 @@
+ #include <stdlib.h>
+ #include <stdio.h>
+ #include <ctype.h>
++#include <sys/types.h>
+ #include "clif.h"
+ #include "dcbtool.h"
+ #include "lldp_dcbx_cmds.h"
+diff --git a/lldptool_cmds.c b/lldptool_cmds.c
+index daef8c8..c793e34 100644
+--- a/lldptool_cmds.c
++++ b/lldptool_cmds.c
+@@ -27,6 +27,7 @@
+ #include <stdio.h>
+ #include <stdlib.h>
+ #include <ctype.h>
++#include <sys/types.h>
+ #include "clif.h"
+ #include "dcb_types.h"
+ #include "lldptool.h"
+-- 
+2.14.3
+
diff --git a/package/open-lldp/0004-fixed-unknown-type-pid_t.patch b/package/open-lldp/0004-fixed-unknown-type-pid_t.patch
deleted file mode 100644
index 2d3b914..0000000
--- a/package/open-lldp/0004-fixed-unknown-type-pid_t.patch
+++ /dev/null
@@ -1,41 +0,0 @@
-From 41e9e6b5aa13d65480cc960a7f15f97d74d64090 Mon Sep 17 00:00:00 2001
-From: Laurent Charpentier <laurent_pubs@yahoo.com>
-Date: Mon, 22 Jan 2018 10:49:45 +0100
-Subject: [PATCH] fixed unknown type pid_t
-
-Fixed 'unknown type pid_t' gcc compile error in dcbtool_cmds.c and
-lldptool_cmds.c
-
-Signed-off-by: Laurent Charpentier <laurent_pubs@yahoo.com>
----
- dcbtool_cmds.c  | 1 +
- lldptool_cmds.c | 1 +
- 2 files changed, 2 insertions(+)
-
-diff --git a/dcbtool_cmds.c b/dcbtool_cmds.c
-index a5cd0fe..ee5c144 100644
---- a/dcbtool_cmds.c
-+++ b/dcbtool_cmds.c
-@@ -27,6 +27,7 @@
- #include <stdlib.h>
- #include <stdio.h>
- #include <ctype.h>
-+#include <sys/types.h>
- #include "clif.h"
- #include "dcbtool.h"
- #include "lldp_dcbx_cmds.h"
-diff --git a/lldptool_cmds.c b/lldptool_cmds.c
-index daef8c8..c793e34 100644
---- a/lldptool_cmds.c
-+++ b/lldptool_cmds.c
-@@ -27,6 +27,7 @@
- #include <stdio.h>
- #include <stdlib.h>
- #include <ctype.h>
-+#include <sys/types.h>
- #include "clif.h"
- #include "dcb_types.h"
- #include "lldptool.h"
--- 
-2.14.3
-
diff --git a/package/open-lldp/Config.in b/package/open-lldp/Config.in
index b40bff8..5bf363e 100644
--- a/package/open-lldp/Config.in
+++ b/package/open-lldp/Config.in
@@ -10,7 +10,7 @@ config BR2_PACKAGE_OPEN_LLDP
 	  configuration tool for Intel LLDP Agent with Enhanced Ethernet
 	  support for the Data Center.
 
-	  http://open-lldp.org/
+	  https://github.com/intel/openlldp/
 
 comment "open-lldp needs a toolchain w/ dynamic library, threads"
 	depends on BR2_STATIC_LIBS || !BR2_TOOLCHAIN_HAS_THREADS
diff --git a/package/open-lldp/open-lldp.hash b/package/open-lldp/open-lldp.hash
index fdd85fe..c3a99c5 100644
--- a/package/open-lldp/open-lldp.hash
+++ b/package/open-lldp/open-lldp.hash
@@ -1,3 +1,3 @@
 # Locally computed
-sha256 f2c8a3307970374b01ad832ea18cf6d9abce6f00b3d8aed9305aa1428ad8252d  open-lldp-036e314bd93602f7388262cc37faf8b626980af1.tar.gz
-sha256 67af5abeb092213cd9c53781503320f635ba28b641e0c3f24d367e8e93a9839b  COPYING
+sha256  112c26241b3fc01496ce4affda4c870122c5343b0505f0a044445ee8056a2106  open-lldp-b71bfb87fefb31c4b1a6a7ae351791c90966c3a8.tar.gz
+sha256  67af5abeb092213cd9c53781503320f635ba28b641e0c3f24d367e8e93a9839b  COPYING
diff --git a/package/open-lldp/open-lldp.mk b/package/open-lldp/open-lldp.mk
index c9fd958..efaaa7b 100644
--- a/package/open-lldp/open-lldp.mk
+++ b/package/open-lldp/open-lldp.mk
@@ -4,8 +4,8 @@
 #
 ################################################################################
 
-OPEN_LLDP_VERSION = 036e314bd93602f7388262cc37faf8b626980af1
-OPEN_LLDP_SITE = git://open-lldp.org/open-lldp.git
+OPEN_LLDP_VERSION = b71bfb87fefb31c4b1a6a7ae351791c90966c3a8
+OPEN_LLDP_SITE = $(call github,intel,openlldp,$(OPEN_LLDP_VERSION))
 OPEN_LLDP_DEPENDENCIES = readline libnl libconfig host-pkgconf
 OPEN_LLDP_LICENSE = GPL-2.0
 OPEN_LLDP_LICENSE_FILES = COPYING
-- 
2.7.4

